packages/sdk-middleware-auth/test/base-auth-flow.spec.js

```diff
@@ -371,6 +371,53 @@ describe('Base Auth Flow', () => {
       createBaseMiddleware({ pendingTasks, tokenCache, requestState }, next)
     }))
 
+  test('should change requestState to false when createAuthMiddlewareBase fails', async () => {
+    const middlewareOptions = createTestMiddlewareOptions()
+    let requestCount = 0
+
+    nock(middlewareOptions.host)
+      .persist() // <-- use the same interceptor for all requests
+      .log(() => {
+        requestCount += 1
+      }) // keep track of the request count
+      .filteringRequestBody(/.*/, '*')
+      .post('/oauth/token', '*')
+      .delay(2000) // <-- Delay the response
+      .reply(401, {
+        message: 'invalid_client',
+        error: 'invalid_client',
+      })
+
+    const requestState = store(false)
+    const pendingTasks = []
+
+    const startCreateBaseMiddleware = () =>
+      new Promise((resolve, reject) => {
+        expect(requestState.get()).toBe(false)
+
+        // fire off promise returning function and change requestState to true
+        // we do not await the response before the assertion since we set
+        // reqeustState to false in the catch block
```

Comment:
nitpick: I checked the code now and technically we don't set it in the catch block but here: https://github.com/commercetools/nodejs/blob/acf7cd4a4b208736dee67a6b946a8e7ccbd07eb8/packages/sdk-middleware-auth/src/base-auth-flow.js#L92-L107